receiver/dcgmreceiver/client.go

```diff
@@ -60,47 +63,46 @@ var dcgmInit = func(args ...string) (func(), error) {
 var dcgmGetLatestValuesForFields = dcgm.GetLatestValuesForFields
 
 func newClient(config *Config, logger *zap.Logger) (*dcgmClient, error) {
-	deviceIndices, names, UUIDs, err := discoverDevices(logger)
+	dcgmCleanup, err := initializeDcgm(config, logger)
 	if err != nil {
-		return nil, err
+		return nil, errors.Join(ErrDcgmInitialization, err)
 	}
-
-	deviceGroup, err := createDeviceGroup(logger, deviceIndices)
-	if err != nil {
-		return nil, err
-	}
-
+	deviceIndices := make([]uint, 0)
+	names := make([]string, 0)
+	UUIDs := make([]string, 0)
+	enabledFieldGroup := dcgm.FieldHandle{}
 	requestedFieldIDs := discoverRequestedFieldIDs(config)
 	supportedFieldIDs, err := getAllSupportedFields()
 	if err != nil {
 		// If there is error querying the supported fields at all, let the
 		// receiver collect basic metrics: (GPU utilization, used/free memory).
 		logger.Sugar().Warnf("Error querying supported profiling fields on '%w'. GPU profiling metrics will not be collected.", err)
 	}
+
 	enabledFields, unavailableFields := filterSupportedFields(requestedFieldIDs, supportedFieldIDs)
 	for _, f := range unavailableFields {
 		logger.Sugar().Warnf("Field '%s' is not supported. Metric '%s' will not be collected", dcgmIDToName[f], dcgmNameToMetricName[dcgmIDToName[f]])
 	}
-	// If there is no metrics to watch after filtering the user configuration
-	// with device supported fields, return an client with an empty device list
-	if len(enabledFields) == 0 {
-		return &dcgmClient{
-			logger:                         logger.Sugar(),
-			enabledFieldIDs:                enabledFields,
-			enabledFieldGroup:              dcgm.FieldHandle{},
-			deviceIndices:                  make([]uint, 0),
-			devicesModelName:               make([]string, 0),
-			devicesUUID:                    make([]string, 0),
-			deviceMetricToFailedQueryCount: make(map[string]uint64, 0),
-		}, nil
-	}
-	enabledFieldGroup, err := setWatchesOnEnabledFields(config, logger, deviceGroup, enabledFields)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to set field watches on %w", err)
+
+	if len(enabledFields) != 0 {
+		deviceIndices, names, UUIDs, err = discoverDevices(logger)
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
We should be consistent about blank lines between these conditionals — I'd remove this one or add one after line 96. Also in line 81.

Reply:
Done!